crates/rslint_cli/src/panic_hook.rs

```diff
@@ -1,23 +1,30 @@
 //! The custom panic hook used by the linter to issue a more descriptive explanation.
 
 use crate::{lint_err, lint_note};
+use std::io::{stderr, Write};
 use std::panic::PanicInfo;
 
 pub fn panic_hook(info: &PanicInfo) {
     lint_err!("The linter panicked unexpectedly. this is a bug.");
```

Comment:
Something i forgot to mention is that this needs to be changed to manually emit and create the diagnostic too, since it uses termcolor stderr (or stdout, cant remember) by default. So we need to make a termcolor [`StandardStreamLock`](https://docs.rs/termcolor/1.1.0/termcolor/struct.StandardStreamLock.html) and emit the diagnostic to it and then write additional info

Reply:
OR
we could use `colored`, make a colored string and write the string to the lock. That would be a lot simpler and nicer and i think it should work.

Reply:
Something akin to what is done in `lib.rs`
https://github.com/RDambrosio016/RSLint/blob/ef02f93768c096293a26bb9c5772f48f527a6b83/crates/rslint_cli/src/lib.rs#L188-L194

Basically
```rust
let mut lock = termcolor::StandardStream::stderr(Always).lock();
```

Reply:
Yep

Reply:
Although it should reuse the lock instead of locking and unlocking

Reply:
So I will have to "unmacro" `lint_err` here. Could it be replaced by a simple `eprintln` or is there some more complexity.

Reply:
Yeah you can swap both of them out for simple writes, #16 adds the `colored` dependency so after that we can simply use that later on.